dotCMS/src/main/java/com/dotmarketing/business/cache/provider/h22/H22Cache.java

```diff
@@ -243,10 +242,32 @@ public void remove(final String group, final String key) {
 		
 	}
 
+	/**
+	 * Calculates the thread allocation % for a given queue size.
+	 * Then determines if that allocation % exceeds or not a tolerance.
+	 * @return
+	 */
+	boolean isAllocationWithinTolerance() {
+		final int size = asyncTaskQueue.size();
+		final float allocation = (float) size / (float) asyncTaskQueueSize;
+		Logger.debug(H22Cache.class,
+				() -> " size is " + size + ", allocation is " + allocation + ", tolerance is :"
+						+ threadAllocationTolerance);
+		return allocation < threadAllocationTolerance;
+	}
 
-    void removeAsync(final Fqn fqn) {
-
+    /**
+     * returns true if async set to true and the task queue is < than a given tolerance % full.
+     *
+     * @return
+     */
+    boolean shouldAsync() {
+       synchronized (H22Cache.class) {
```

Comment:
We don't need the sync block here, I don't think.

Reply:
ok. removed